src/newmatrix.cpp

```diff
@@ -1738,34 +1735,40 @@ ACMD(do_logon)
       send_to_icon(PERSONA, "This host is not connected to a RTG.\r\n");
       return;
     }
-  } else if (!str_cmp(argument, "access") && matrix[PERSONA->in_host].stats[ACCESS][MTX_STAT_TRAPDOOR])
-    target_host = real_host(matrix[PERSONA->in_host].stats[ACCESS][MTX_STAT_TRAPDOOR]);
-  else if (!str_cmp(argument, "control") && matrix[PERSONA->in_host].stats[CONTROL][MTX_STAT_TRAPDOOR])
-    target_host = real_host(matrix[PERSONA->in_host].stats[CONTROL][MTX_STAT_TRAPDOOR]);
-  else if (!str_cmp(argument, "index") && matrix[PERSONA->in_host].stats[INDEX][MTX_STAT_TRAPDOOR])
-    target_host = real_host(matrix[PERSONA->in_host].stats[INDEX][MTX_STAT_TRAPDOOR]);
-  else if (!str_cmp(argument, "files") && matrix[PERSONA->in_host].stats[FILES][MTX_STAT_TRAPDOOR])
-    target_host = real_host(matrix[PERSONA->in_host].stats[FILES][MTX_STAT_TRAPDOOR]);
-  else if (!str_cmp(argument, "slave") && matrix[PERSONA->in_host].stats[SLAVE][MTX_STAT_TRAPDOOR])
-    target_host = real_host(matrix[PERSONA->in_host].stats[SLAVE][MTX_STAT_TRAPDOOR]);
-  else {
-    for (struct exit_data *exit = matrix[PERSONA->in_host].exit; exit; exit = exit->next)
+  } else {
+    // Subsystem trap door? Test against the local subsystem
+    for (int sub = ACCESS; (target_host < 0) && (sub < NUM_OF_SUBSYSTEMS); sub++) {
```

Comment:
What's the purpose of the `target_host < 0` block? If it's to terminate iteration on a successful if-check, this is better done with an explicit `break` statement.

Reply:
Done